app/validation/validator.py

```diff
@@ -281,6 +282,21 @@ def _validate_answer_actions(self, answer):
 
         return errors
 
+    def _ensure_answer_labels_and_values_match(self, answer):
+        errors = []
+        for option in answer.get('options', []):
+            if isinstance(option['label'], str):
+                if option['label'] != option['value']:
+                    errors.append(
+                        (self._error_message(f'`Found mismatching answer value for label: {option["label"]} '
+                                             f'in answer id: {answer["id"]}`')))
+            else:
+                if option['label']['text'] != option['value']:
+                    errors.append(
+                        (self._error_message(f'`Found mismatching answer value for label: {option["label"]["text"]} '
+                                             f'in answer id: {answer["id"]}`')))
```

Comment:
This could be simplified to:

```python
if isinstance(option['label'], str):
    label = option['label']
else:
    label = option['label']['text']
           
if label != option['value']:
    errors.append(
        (self._error_message(f'`Found mismatching answer value for label: {label} '
                             f'in answer id: {answer["id"]}`')))
```

Reply:
Refactored this